### PR TITLE
Move out removeForChildWindow() from resources-impl.js

### PR DESCRIPTION
**Pull request by lannka**
For #23311

### CODE REVIEW

File: src/friendly-iframe-embed.js [hunk 1]
```diff
@@ -550,6 +550,21 @@ export class FriendlyIframeEmbed {
     );
   }
 
+  /**
+   * Removes all resources belonging to the FIE window.
+   * @private
+   */
+  removeResources_() {
```
Comment (dvoytenko) [inline src/friendly-iframe-embed.js]: IMHO, a much better option is when a FIE ampdoc gets its own Resources instance that can be completely thrown out.

Comment (lannka) [reply src/friendly-iframe-embed.js]: Agree. This PR still brings value though, as once this `childwindow` concept is moved out of resources service, we don't need to worry about providing a dummy impl for inabox runtime.

**Comment by mrjoro** [comment]
Is this PR still active?

File: src/friendly-iframe-embed.js (hunk 1 above)
Comment (lannka) [reply src/friendly-iframe-embed.js]: @dvoytenko this is a pure dependency clean up, I think we should get it in. No?

**Comment by lannka** [comment]
```
The bundle size (gzipped compressed size of v0.js) of this pull request could not be determined because the base size (that is, the bundle size of the master commit that this pull request was compared against) was not found in the https://github.com/ampproject/amphtml-build-artifacts repository. This can happen due to failed or delayed Travis builds on said master commit.
```

I retried the whole travis build, it failed with same msg. @danielrozenberg have you seen this before?

**Comment by danielrozenberg** [comment]
Yes, this is unfortunately a side-effect of master being red. Since this PR changes `src/` files, I can't rubberstamp it for bundle-size. You can wait until master is green again (you can help hasten this by approving #24415) and rebase your PR on that master

**Comment by lannka** [comment]
@danielrozenberg that worked! thanks

**Pull request merged**